src/components/link_teaser/index.tsx

```diff
@@ -0,0 +1,86 @@
+import Image from '@/components/image'
+import { get } from 'lodash'
+import { recordEvent } from '@/utils/recordEvent'
+import { ParagraphLinkTeaser } from '@/types/paragraph'
+
+export interface LinkTeaserProps extends ParagraphLinkTeaser {
```

Comment:
i think we want to use this `ParagraphLinkTeaser` in the data service and not in the "pure" component. If the component props are still extending this it's going to expect all the drupal stuff along with it, isn't it? 

something like:

```js
export const transformLinkTeaserData = function (
  entity: ParagraphLinkTeaser,
  viewMode: string
): LinkTeaserProps { 
  // type check the entity is the right paragraph here
 
  return {
    ...mapping to the "pure" component props here
  }
}
```

Reply:
Ok great, I'll fix in the morning, I have a question for you about the componentParams type definitions

Reply:
Fixed!